src/widgets/dialogs/SelectChannelDialog.cpp

```diff
@@ -36,159 +31,138 @@ SelectChannelDialog::SelectChannelDialog(QWidget *parent)
-                         [=](bool enabled) mutable {
-                             live_lbl->setVisible(enabled);
-                         });
-
-        // automod_btn
-        auto automod_btn = vbox.emplace<QRadioButton>("AutoMod").assign(
-            &this->ui_.twitch.automod);
-        auto automod_lbl =
-            vbox.emplace<QLabel>("Shows when AutoMod catches a message in any "
-                                 "channel you moderate.")
-                .hidden();
-
-        automod_lbl->setWordWrap(true);
-        automod_btn->installEventFilter(&this->tabFilter_);
-
-        QObject::connect(automod_btn.getElement(), &QRadioButton::toggled,
-                         [=](bool enabled) mutable {
-                             automod_lbl->setVisible(enabled);
-                         });
-
-        vbox->addStretch(1);
-
-        // tabbing order
-        QWidget::setTabOrder(automod_btn.getElement(),
-                             channel_btn.getElement());
-        QWidget::setTabOrder(channel_btn.getElement(),
-                             whispers_btn.getElement());
-        QWidget::setTabOrder(whispers_btn.getElement(),
-                             mentions_btn.getElement());
-        QWidget::setTabOrder(mentions_btn.getElement(),
-                             watching_btn.getElement());
-        QWidget::setTabOrder(watching_btn.getElement(), live_btn.getElement());
-        QWidget::setTabOrder(live_btn.getElement(), automod_btn.getElement());
-
-        // tab
-        auto *tab = notebook->addPage(obj.getElement());
-        tab->setCustomTitle("Twitch");
-    }
-
-    layout->setStretchFactor(notebook.getElement(), 1);
-
-    auto buttons =
-        layout.emplace<QHBoxLayout>().emplace<QDialogButtonBox>(this);
-    {
-        auto *button_ok = buttons->addButton(QDialogButtonBox::Ok);
-        QObject::connect(button_ok, &QPushButton::clicked, [this](bool) {
-            this->ok();
-        });
-        auto *button_cancel = buttons->addButton(QDialogButtonBox::Cancel);
-        QObject::connect(button_cancel, &QAbstractButton::clicked,
-                         [this](bool) {
-                             this->close();
-                         });
-    }
+    auto *layout = new QVBoxLayout(this->getLayoutContainer());
 
-    this->setMinimumSize(300, 310);
-    this->ui_.notebook->selectIndex(TAB_TWITCH);
-    this->ui_.twitch.channel->setFocus();
+    auto &ui = this->ui_;
+    // Channel
+    ui.channel = new AutoCheckedRadioButton("Channel");
+    layout->addWidget(ui.channel);
+
+    ui.channelLabel = new QLabel("Join a Twitch channel by its channel name");
+    ui.channelLabel->setVisible(false);
+    layout->addWidget(ui.channelLabel);
+
+    ui.channelName = new QLineEdit();
+    ui.channelName->setVisible(false);
+    layout->addWidget(ui.channelName);
+
+    QObject::connect(ui.channel, &AutoCheckedRadioButton::toggled, this,
+                     [this](bool enabled) mutable {
```

Comment:
```suggestion
                     [this](bool enabled) {
```

Same for the other cases.

Reply:
not part of this PR but resolved in [`8d02c53` (#6081)](https://github.com/Chatterino/chatterino2/pull/6081/commits/8d02c535036ded4a43deaa3db2fafa3a01b2302e)